modules/compiler/utils/source/cl_builtin_info.cpp

```diff
@@ -2820,13 +2840,66 @@ static multi_llvm::Optional<unsigned> parseMemoryOrderParam(Value *const P) {
   return multi_llvm::None;
 }
 
-Instruction *CLBuiltinInfo::mapSyncBuiltinToMuxSyncBuiltin(
+Instruction *CLBuiltinInfo::lowerBuiltinToMuxBuiltin(
     CallInst &CI, BIMuxInfoConcept &BIMuxImpl) {
   auto &M = *CI.getModule();
   auto *const F = CI.getCalledFunction();
   assert(F && "No calling function?");
   auto const ID = identifyBuiltin(*F);
 
+  // Handle straightforward 1:1 mappings.
+  if (auto MuxID = [](BuiltinID ID) -> std::optional<BuiltinID> {
```

Comment:
I think I'd find this more readable outside the if (). Also may be worth factoring it out as this function is pretty big.

Reply:
Yeah I agree, I've split that into its own function.